cclib/method/ddec.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2020, the cclib development team
+#
+# This file is part of cclib (http://cclib.github.io) and is distributed under
+# the terms of the BSD 3-Clause License.
+
+"""Calculation of DDEC charges based on data parsed by cclib."""
+import copy
+import random
+import numpy
+import logging
+import math
+import os
+import sys
+
+from cclib.method.calculationmethod import Method
+from cclib.method.volume import electrondensity_spin
+from cclib.parser.utils import convertor
+from cclib.parser.utils import find_package
+
+
+class MissingInputError(Exception):
+    pass
+
+
+class DDEC6(Method):
+    """DDEC6 charges."""
+
+    # All of these are required for DDEC6 charges.
+    required_attrs = ("homos", "mocoeffs", "nbasis", "gbasis")
+
+    def __init__(
+        self, data, volume, proatom_path=None, progress=None, loglevel=logging.INFO, logname="Log"
+    ):
+        super(DDEC6, self).__init__(data, progress, loglevel, logname)
+
+        self.volume = volume
+        self.fragresults = None
+
+        if numpy.sum(self.data.coreelectrons) != 0:
+            # TODO: Pseudopotentials should be added back
+            pass
+
+        assert os.path.isdir(
+            proatom_path
+        ), "Directory that contains chargemol proatom densities should be added as an input."
+        # Read in reference charges
```

Comment:
I think this comment applies to lines up to 56 in current numbering. Perhaps add a line before the comment, and eliminate the blank lines in this block, for readability.

Reply:
I've added a comment & reflowed the code for better readability. Thanks!